corehq/apps/reports/standard/deployments.py

```diff
@@ -209,8 +210,33 @@ def user_query(self, pagination=True):
                                        )
         return user_query
 
+    def get_location_columns(self, grouped_ancestor_locs):
+        from corehq.apps.locations.models import LocationType
+        location_types = LocationType.objects.by_domain('icds-cas')
+        all_user_locations = grouped_ancestor_locs.values()
+        all_user_loc_types = {loc.location_type_id for user_locs in all_user_locations for loc in user_locs}
+        required_loc_columns = [loc_type for loc_type in location_types if loc_type.id in all_user_loc_types]
+        return required_loc_columns
+
+    def user_locations(self, ancestors, location_type):
+        location_list = ['---'] * len(location_type)
+        types_id = [type.id for type in location_type]
+        for loc in ancestors:
+            position = types_id.index(loc.location_type_id)
+            location_list[position] = loc.name
+        return location_list
+
     def process_rows(self, users, fmt_for_export=False):
         rows = []
+        users = list(users)
```

Comment:
How big can this list get?  I'd try out the query on a realistic set to see how it works - I wouldn't be surprised if this needs to be broken into chunks.

Reply:
this can go to around 400K
just one point 
`users = list(user)` and 
`iterating user using a for loop`.  is there a difference in time consumption of two. Or is loop is better in any sense. Because  below it was already using loop. and performing really good.

Reply:
I was thinking more about passing this to the `SQLLocation` query - this will result in a query with a very large list of parameters, and I'm not sure how that'll do.

Reply:
It have tested it running on scale data for around 400K location list. The query perform reasonable good. In fact it was again faster than using chunks and query the database multiple times.